pkg/distributor/field_detection_test.go

```diff
@@ -167,6 +167,106 @@ func Test_DetectLogLevels(t *testing.T) {
 			Value: constants.LogLevelTrace,
 		})
 	})
+
+	t.Run("detected_level with uppercase value gets normalized to lowercase", func(t *testing.T) {
+		limits, ingester := setup(true)
+		distributors, _ := prepare(t, 1, 5, limits, func(_ string) (ring_client.PoolClient, error) { return ingester, nil })
+
+		// Create a write request with detected_level in uppercase
+		writeReq := makeWriteRequestWithLabels(1, 10, []string{`{foo="bar"}`}, false, false, false)
+		writeReq.Streams[0].Entries[0].Line = `some log message`
+		writeReq.Streams[0].Entries[0].StructuredMetadata = push.LabelsAdapter{
+			{
+				Name:  constants.LevelLabel, // detected_level
+				Value: "ERROR",              // Uppercase value
+			},
+		}
+
+		_, err := distributors[0].Push(ctx, writeReq)
+		require.NoError(t, err)
+		topVal := ingester.Peek()
+		require.Equal(t, `{foo="bar"}`, topVal.Streams[0].Labels)
+
+		// Verify that detected_level is normalized to lowercase
+		sm := topVal.Streams[0].Entries[0].StructuredMetadata
+		require.Len(t, sm, 1)
+		require.Equal(t, constants.LevelLabel, sm[0].Name)
+		require.Equal(t, constants.LogLevelError, sm[0].Value) // Should be lowercase "error"
+	})
+
+	t.Run("detected_level with mixed case value gets normalized to lowercase", func(t *testing.T) {
```

Comment:
What is the difference to the previous test case?

Reply:
good point `ERROR` should just become a case in this test.